src/services/healthcheck.service.js

```diff
@@ -0,0 +1,112 @@
+import { config as hathorLibConfig, healthApi, txMiningApi } from '@hathor/wallet-lib';
+
+const { buildComponentHealthCheck } = require('../helpers/healthcheck.helper');
+const { friendlyWalletState } = require('../helpers/constants');
+
+const healthService = {
+  /**
+   * Returns the health object for a specific wallet
+   *
+   * @param {HathorWallet} wallet
+   * @param {string} walletId
+   * @returns {Object}
+   */
+  async getWalletHealth(wallet, walletId) {
+    let healthData;
+
+    if (!wallet.isReady()) {
+      healthData = buildComponentHealthCheck(
+        `Wallet ${walletId}`,
+        'fail',
+        'internal',
+        `Wallet is not ready. Current state: ${friendlyWalletState[wallet.state]}`
+      );
+    } else {
+      healthData = buildComponentHealthCheck(
+        `Wallet ${walletId}`,
+        'pass',
+        'internal',
+        'Wallet is ready'
+      );
+    }
+
+    return healthData;
+  },
+
+  /**
+   * Returns the health object for the connected fullnode
+   *
+   * @returns {Object}
+   */
+  async getFullnodeHealth() {
+    let output;
+    let healthStatus;
+
+    // TODO: We will need to parse the healthData to get the status,
+    // but hathor-core hasn't this implemented yet
+    try {
+      await healthApi.getHealth();
```

Comment:
Do we have it already?

Reply:
No, in the next release of hathor-core. But I believe it's in the verge of being released.